spec/dummy_models_total_count_controller_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'spec_helper'
+
+describe DummyModelsTotalCountController, type: :controller do
+  describe '#index' do
+    let!(:dummy_models) { create_list(:dummy_model, 28) }
+    let(:expected_list) do
+      dummy_models.first(25).map do |dummy|
+        { 'id' => dummy.id, 'name' => dummy.name, 'something' => dummy.something }
+      end
+    end
+
+    context 'when paginating an ActiveRecord with no previous pagination but kaminari installed' do
+      before do
+        get :index
+      end
+
+      include_examples 'total count pagination param'
+
+      it 'responds with valid page' do
+        expect(response_body(response)['page']).to eq expected_list
+      end
```

Comment:
What do you think to add here and expectation for `DummyModel.count` not to have been called

Reply:
The shared example includes an expectation for `DummyModel.count` that checks if it's different from the ovewritten `total_count` option (to prove that the `total_count` becomes into the value passed through the option and not into the `DummyModel.count` itself). As said in a comment before, this is just for the `total_count` param, which actually counts all the collection to be rendered.